### PR TITLE
feat(dynamic facets): facet ordering support

### PR DESCRIPTION
**Pull request by aallam**
| Q                 | A
| ----------------- | ----------
| Bug fix?          | no
| New feature?      | yes
| BC breaks?        | no     
| Need Doc update   | yes

## Describe your change

The new Facet Ordering feature providing the order of facet widgets and facet values.

### CODE REVIEW

File: algoliasearch-core/src/main/java/com/algolia/search/models/indexing/SearchResult.java [hunk 1]
```diff
@@ -131,6 +133,15 @@ public String getQueryAfterRemoval() {
     return this;
   }
 
+  public RenderingContent getRenderingContent() {
+    return renderingContent;
+  }
+
+  public SearchResult<T> setRenderingContent(RenderingContent renderingContent) {
```
Comment (Haroenv) [inline algoliasearch-core/src/main/java/com/algolia/search/models/indexing/SearchResult.java:140]: I don't understand why this needs a public setter, but the other items seem to have it, so it's ok

File: algoliasearch-core/src/main/java/com/algolia/search/models/rules/FacetValuesOrder.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  public List<String> getOrder() {
+    return order;
+  }
+
+  public FacetValuesOrder setOrder(List<String> order) {
+    this.order = order;
+    return this;
+  }
+
+  public String getSortRemainingBy() {
```
Comment (Haroenv) [inline algoliasearch-core/src/main/java/com/algolia/search/models/rules/FacetValuesOrder.java:30]: can this be more restricted than string?

File: algoliasearch-core/src/main/java/com/algolia/search/models/rules/Consequence.java [hunk 3]
```diff
@@ -60,9 +60,19 @@ public Consequence setFilterPromotes(Boolean filterPromotes) {
...
+    this.renderingContent = renderingContent;
+    return this;
+  }
+
   private ConsequenceParams params;
   private List<ConsequencePromote> promote;
   private Map<String, Object> userData;
   private List<Hide> hide;
   private Boolean filterPromotes;
+  private RenderingContent renderingContent;
```
Comment (Haroenv) [inline algoliasearch-core/src/main/java/com/algolia/search/models/rules/Consequence.java]: renderingContent is actually _inside_ params in the consequence

File: algoliasearch-core/src/main/java/com/algolia/search/models/indexing/SearchResult.java (hunk 1 above)
Comment (aallam) [reply algoliasearch-core/src/main/java/com/algolia/search/models/indexing/SearchResult.java:140]: it is required by the serialization library (jackson)

File: algoliasearch-core/src/main/java/com/algolia/search/models/rules/FacetValuesOrder.java (hunk 2 above)
Comment (aallam) [reply algoliasearch-core/src/main/java/com/algolia/search/models/rules/FacetValuesOrder.java:30]: We can, but this client's philosophy avoids introducing restrictions (this is observable in many other places in the codebase)

File: algoliasearch-core/src/main/java/com/algolia/search/models/rules/Consequence.java (hunk 3 above)
Comment (aallam) [reply algoliasearch-core/src/main/java/com/algolia/search/models/rules/Consequence.java]: Nice catch! fixed :)

**Review by Haroenv — APPROVED** [review]
looks consistent!

**Pull request merged**